### DISCUSSION TITLE
Pingvin Share - Change data directory

### DISCUSSION THREAD

**Discussion by dhenry437** [tteck/Proxmox#2844] [Help Needed] [answered]
Hi,

I have spun up a Pingvin Share LXC via the script, and I cant find anything in the Pingvin docs about how to change the data directory other than an environment variable that should be set to the desired directory at the time of installation.

Does anyone know how to change the data directory after the fact? Could it be as simple as setting the `DATA_DIRECTORY` environment variable and running the update command?

Thanks
Dan

**Comment by dhenry437** [comment, marked as answer]
Finally solved with a PM2 config file. 

#### ecosystem.config.js

``` js
const dataDir = "/mnt/media/pingvin"

module.exports = {
  apps : [{
    name: "pingvin-share-backend",
    script: "npm",
    args: "run prod",
    cwd: "/opt/pingvin-share/backend",
    env: {
      DATA_DIRECTORY: dataDir,
    },
  }, {
    name: "pingvin-share-frontend",
    script: "npm",
    args: "run start",
    cwd: "/opt/pingvin-share/frontend",
    env: {
      DATA_DIRECTORY: dataDir,
    },
  }]
}
```

This approach could also be used to set any of the other variables like `PORT` or `DATABASE_URL`.

Don't know enough about bash to make a proper pull request but I think you would remove the two `$STD pm2 start ...` lines (44 & 48 at the time of writing) from `pingvin-install.sh` and put something like `pm2 start ecosystem.config.js` (will change depending on where you want the ecosystem file to live) before `pm2 startup`.

Just a side note, I thought you needed a `pm2 save` after `pm2 startup` but the script is working fine with pm2 persisting after reboot so I guess I need to read into it more.

**Reply by ucramos** [reply]
Hi,

Thank you for the script.
When i reboot my LXC it stops working. I have to "pm2 start ecosystem.config.js" on every boot.
How did you set it to run at start?

**Reply by dhenry437** [reply]
https://pm2.keymetrics.io/docs/usage/startup/

You'll need to `pm2 start ecosystem.config.js` then `pm2 save` 

More information at the link

**Reply by ucramos** [reply]
Worked like a charm.

Thank You!

**Reply by tteck** [reply]
Do I need to add `pm2 save` to the script?
